thrift/thrift-gen/main.go

```diff
@@ -83,15 +106,43 @@ func processFile(generateThrift bool, inputFile string, outputFile string) error
 		return fmt.Errorf("Could not parse .thrift file: %v", err)
 	}
 
-	goTmpl := parseTemplate()
-	for filename, v := range parsed {
-		if err := generateCode(outputFile, goTmpl, packageName(filename), v); err != nil {
+	goTmpl := parseTemplate() // tchan-template
+	for thriftFilename, AST := range parsed {
+		if err := generateCode(outputFile, goTmpl, packageName(thriftFilename), AST); err != nil {
 			return err
 		}
-		// TODO(prashant): Support multiple files / includes etc?
-		return nil
-	}
 
+		templates, err := getTemplatesFromFiles(templateFiles, AST)
+		if err != nil {
+			return fmt.Errorf("Could not parse thrift-gen file: %v", err)
+		}
+
+		for templateFilename, template := range templates {
+			buf := &bytes.Buffer{}
+			buf.WriteString("// @generated Code generated by thrift-gen. Do not modify.\n\n")
+
+			if err := template.Execute(buf, &FileBasedTemplateDate{
+				Package: packageName(thriftFilename),
+				AST:     AST,
+				State:   NewState(AST),
+			}); err != nil {
+				return fmt.Errorf("Could not parse thrift-gen template: %v", err)
```

Comment:
lower case error message, and avoid "thrift-gen" in the error, I think just "could not parse template: %v" is fine.